### PR TITLE
Implement MethodHandles.dropReturn for JDK16

### PR DESCRIPTION
**Pull request by babsingh**
Related: https://github.com/eclipse/openj9/issues/11135

Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

**Comment by babsingh** [comment]
fyi @pshipton @tajila

File: jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java [hunk 1]
```diff
@@ -5596,7 +5596,45 @@ static MethodHandle collectReturnValue(MethodHandle target, MethodHandle filter)
...
+	 * @throws NullPointerException if target is null.
+	 */
+	public static MethodHandle dropReturn​(MethodHandle target) {
+		/* Implicit null check on target. */
+		MethodType targetType = target.type();
+		if (targetType.returnType() == void.class) {
+			return target;
+		}
+		targetType = targetType.changeReturnType(void.class);
+		return buildTransformHandle(new DropReturnHelper(target), targetType);
```
Comment (DanHeidinga) [inline jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java:5634]: I don't think we need this new helper class.  It should be possible to wrap the `target` with a `FilterReturnHandle` to remove the return value.

See the existing code in MethodHandle#asType

Comment (babsingh) [reply jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java]: Updated. I had initially thought of using `FilterReturnHandle` but I was stuck at creating the `filter` MH. Thanks for pointing me to the example.

**Review by DanHeidinga — APPROVED** [review]
lgtm

**Comment by DanHeidinga** [comment]
Jenkins test sanity xlinux jdk16

**Comment by babsingh** [comment]
[Build_JDK16_x86-64_linux_Personal](https://ci.eclipse.org/openj9/job/Build_JDK16_x86-64_linux_Personal/1/):
```
15:41:57  /home/jenkins/workspace/Build_JDK16_x86-64_linux_Personal/build/linux-x86_64-server-release/support/j9jcl/java.base/share/classes/java/lang/invoke/MethodHandles.java:5282: error: ';' expected
15:41:57  	public static MethodHandle dropReturn???(MethodHandle target) {
15:41:57  	                                     ^
15:41:57  /home/jenkins/workspace/Build_JDK16_x86-64_linux_Personal/build/linux-x86_64-server-release/support/j9jcl/java.base/share/classes/java/lang/invoke/MethodHandles.java:5282: error: ';' expected
15:41:57  	public static MethodHandle dropReturn???(MethodHandle target) {
15:41:57  	                                                            ^
15:41:57  2 errors
```

`vi` shows a hidden character, which has been removed. `Jenkins test sanity xlinux jdk16` will need to be rerun.

**Comment by DanHeidinga** [comment]
Jenkins test sanity xlinux jdk16

**Pull request merged**